dotrix_core/src/assets/buffer.rs

```diff
@@ -0,0 +1,162 @@
+//! Buffer asset
+//!
+
+use crate::{
+    reloadable::*,
+    renderer::{Buffer as GpuBuffer, Renderer},
+};
+use dotrix_derive::*;
+
+/// Buffer asset
+#[derive(Reloadable, BufferProvider)]
+pub struct Buffer {
+    /// The length of the data that the buffer was created with
+    pub data_len: usize,
+    /// The underlying gpu buffer
+    pub buffer: GpuBuffer,
+    /// Flagged on change
+    pub reload_state: ReloadState,
+}
```

Comment:
[Question]
What about using GpuBuffer itself as an asset? Why we need a wrapper?

Reply:
Was debating this but went with this for kind of seperation of concern. This handles the high level representation, while the Renderer one the low level. Originally this struct behaved a bit like `assets::Texture` in that it held a copy of the data in a `Vec` but there was no real use case for it over just direct loading.

It also made it obvious to `impl BufferProvider` which allows the abstraction of having any struct provide a reference to a `GpuBuffer` for the Pipieline, although I suppose `GpuBuffer` could `impl BufferProvider` and return `&self`